controlplane/controllers/scale.go

```diff
@@ -159,6 +160,29 @@ func (r *CK8sControlPlaneReconciler) scaleDownControlPlane(
 	}
 	**/
 
+	clusterObjectKey := util.ObjectKey(cluster)
+	workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, clusterObjectKey)
+	if err != nil {
+		logger.Error(err, "failed to create client to workload cluster")
+		return ctrl.Result{}, errors.Wrapf(err, "failed to create client to workload cluster")
+	}
+
+	authToken, err := token.Lookup(ctx, r.Client, clusterObjectKey)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to lookup auth token: %w", err)
+	}
+
+	if authToken == nil {
+		return ctrl.Result{}, fmt.Errorf("auth token not yet generated")
+	}
+
+	// TODO: Get from CK8sConfig
+	microclusterPort := 2380
```

Comment:
I could not really figure out what is the best way to access the cluster config from this controller.

Reply:
NOTE: microcluster port is not something that changes, and therefore should be handled in the workload cluster, not in every method.

Reply:
added it to the `GetWorkloadCluster` call